### PR TITLE
fix: 23369 back button navigation fixed for ondify steps

### PR DESCRIPTION
**Pull request by vipin87**
### Fixed Issues

$ https://github.com/Expensify/App/issues/23369
PROPOSAL: https://github.com/Expensify/App/issues/23369#issuecomment-1650293399

### Tests

1. Open the "Settings" menu.
2. Click on "Workspaces."
3. Navigate to "Bank Account."
4. Click on "Connect bank account."
5. Proceed through all the steps until you reach the "Personal Information" section (step 3 of 5).
6. Click on "Save & Continue" in the "Personal Information" section (step 3 of 5).
7. Click on "Drivers License" in the "Personal Information" section (step 3 of 5) and proceed.
8. Click on "Get Secure link" in the "Personal Information" section (step 3 of 5) and proceed.
9. At this point, when you reach the page where "Scan the QR code with your phone" is written, click on the back button.
10. Observe that the user is directed back to the previous page
11. Keep clicking back until first page is reached

- [x] Verify that no errors appear in the JS console

### Offline tests

Onfido flow not available in offline mode. It has to be tested while application is online.

### QA Steps

Same as "Tests" section above

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/32353512/1d9ad021-8437-4c9a-b540-f7b075422d9c

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/32353512/e6d429d4-3a37-4061-945f-222733a9bc3a

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/32353512/5e99b38f-316c-4b07-8a0e-f58463c6a250

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/32353512/baa445fb-bb77-4ac6-88ea-0d50756350e5

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/32353512/be718a51-16d2-42e0-a25c-38d6da90a8a9

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/32353512/c23c457c-0c3a-4651-8f01-98da209700c2

</details>

### CODE REVIEW

**Comment by vipin87** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/pages/ReimbursementAccount/RequestorOnfidoStep.js [hunk 1]
```diff
@@ -31,6 +31,16 @@ class RequestorOnfidoStep extends React.Component {
     constructor(props) {
         super(props);
         this.submit = this.submit.bind(this);
+        this.goBack = this.goBack.bind(this);
+    }
+
+    goBack(...args) {
+        const ondifyBack = document.getElementsByClassName('onfido-sdk-ui-NavigationBar-back');
```
Comment (robertKozik) [inline src/pages/ReimbursementAccount/RequestorOnfidoStep.js:38]: Can't we use `document.querySelector` to get the first occurrence of that class? As we won't be having the array this way code would be simplified

Comment (robertKozik) [reply src/pages/ReimbursementAccount/RequestorOnfidoStep.js:38]: Also typo in the const name `ondify` should be `onfido`

**Review by robertKozik — CHANGES_REQUESTED** [review]
Few things to consider

File: src/pages/ReimbursementAccount/RequestorOnfidoStep.js (hunk 1 above)
Comment (vipin87) [reply src/pages/ReimbursementAccount/RequestorOnfidoStep.js]: @robertKozik yes you are right, fixed now.

File: src/pages/ReimbursementAccount/RequestorOnfidoStep.js [hunk 2]
```diff
@@ -31,6 +31,16 @@ class RequestorOnfidoStep extends React.Component {
     constructor(props) {
         super(props);
         this.submit = this.submit.bind(this);
+        this.goBack = this.goBack.bind(this);
+    }
+
+    goBack(...args) {
```
Comment (robertKozik) [inline src/pages/ReimbursementAccount/RequestorOnfidoStep.js:37]: Didn't saw that one, `onBackButtonPress` is called inside `HeaderWithBackButton` without any args, so this spread is redundant in my opinion

Comment (robertKozik) [reply src/pages/ReimbursementAccount/RequestorOnfidoStep.js:37]: Sorry for this

**Review by robertKozik — CHANGES_REQUESTED** [review]
One minor thing, sorry

File: src/components/Onfido/index.css [hunk 3]
```diff
@@ -47,3 +47,7 @@
         height: 92% !important;
     }
 }
+
+.onfido-sdk-ui-NavigationBar-back {
```
Comment (robertKozik) [inline src/components/Onfido/index.css:51]: if you could add the explanation comment why we are hiding this button

File: src/pages/ReimbursementAccount/RequestorOnfidoStep.js (hunk 2 above)
Comment (vipin87) [reply src/pages/ReimbursementAccount/RequestorOnfidoStep.js]: fixed

File: src/components/Onfido/index.css (hunk 3 above)
Comment (vipin87) [reply src/components/Onfido/index.css:51]: fixed

**Comment by vipin87** [comment]
@robertKozik is there anything still pending on this PR?

**Comment by robertKozik** [comment]
Sorry @vipin87 coming back to this PR now

**Comment by robertKozik** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [X] Android / Chrome
    - [ ] iOS / native
    - [X] iOS / Safari
    - [ X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by robertKozik** [comment]
@vipin87 Could you kindly change the offline test steps, as I believe onfido flow is not accessible while offline? Also please check all the checkboxes in your Author checklist. Thank you in advance

**Comment by vipin87** [comment]
@robertKozik checked all checkboxes and updated offline steps. Sorry for to many to-and-fro, as it is my first PR.

**Comment by robertKozik** [comment]
@vipin87 gentle bump on updating test steps and checkboxes. TYIA

**Comment by vipin87** [comment]
@robertKozik I already did those changes.
image

**Comment by robertKozik** [comment]
Sorry, didn't saw them. I'll proceed with the testing then

**Review by robertKozik — APPROVED** [review]
Looks good, lets push this forward

**Review by francoisl — APPROVED** [review]
LGTM

File: src/pages/ReimbursementAccount/RequestorOnfidoStep.js [hunk 4]
```diff
@@ -31,6 +31,16 @@ class RequestorOnfidoStep extends React.Component {
     constructor(props) {
         super(props);
         this.submit = this.submit.bind(this);
+        this.goBack = this.goBack.bind(this);
+    }
+
+    goBack() {
+        const onfidoBack = document.querySelector('.onfido-sdk-ui-NavigationBar-back');
```
Comment (situchan) [inline src/pages/ReimbursementAccount/RequestorOnfidoStep.js:38]: This caused deploy blocker - https://github.com/Expensify/App/issues/24343.
`document` doesn't exist on native.

Comment (robertKozik) [reply src/pages/ReimbursementAccount/RequestorOnfidoStep.js:38]: Dang, I didn't encountered this during my testing on offline...
@vipin87 Can you rise the follow-up for the issue with proper platform handling?

**Pull request merged**